doc/guides/deploy-without-root.rst

```diff
@@ -0,0 +1,92 @@
+Deploying without Root
+===
+
+Requirements
+---
+
+NixOps 2.0 allows for deploying as users other than root, as long
+as the deploying user meets two requirements:
+
+1. The user can become root without needing to type a password.
+2. Nix considers the user to be a `"trusted user" <https://nixos.org/nix/manual/#conf-trusted-users>`_.
+
+In this guide, we will use passwordless sudo.
+
+This guide assumes:
+1. The deploying user's name is "deployer"
+2. The target machine's name is "hermes"
+3. the target machine is already managed by NixOps
+
+Steps
+---
+
+1. Configure the target machine according to the listed requirements.
+2. Update the NixOps network to use our alternative user.
+3. Deploy as the new user.
+
+
+Configuring the Target Machine
+***
+
+First, mark the deploying user as trusted:
+
+.. code-block:: nix
+  {
+    nix.trustedUsers = [ "deployer" ];
+  }
+
+This will let the user copy Nix store paths to the target.
+
+Let the deploying user use sudo:
+
+.. code-block:: nix
+  {
+    users.users.deployer.extraGroups = [ "wheel" ];
+  }
+
+Then, we configure the machine to have passwordless sudo:
+
+.. code-block:: nix
+  {
+    security.sudo.wheelNeedsPassword = false;
+  }
+
+Now use NixOps to deploy these changes to the server before taking
+the next step.
+
+Configuring the NixOps Network
+***
+
+Edit your network.nix to specify the machine's
+``deployment.targetUser``:
+
+.. code-block:: nix
+
+  {
+    network.description = "Non-root deployment";
+
+    hermes =
+      { resources, ... }:
```

Comment:
Is `resources` necessary here? Or is it because of the contents prior to trimming it?

Reply:
I think it is useful to provide context in to where this goes, like to give a contextual anchor.

Reply:
Sure, makes sense.